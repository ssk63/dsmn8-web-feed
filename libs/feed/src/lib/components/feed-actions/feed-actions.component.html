<div class="feed-actions">
  <dsmn8-button
    *ngIf="likesButtonLabel"
    class="feed-actions__likes-btn"
    [label]="likesButtonLabel"
    [isFullWidth]="true"
    [isPrimary]="true"
    (click)="likesBtnClicked.emit($event)"
  ></dsmn8-button>
  <dsmn8-button
    *ngIf="reshareButtonLabel"
    class="feed-actions__reshare-btn"
    [label]="reshareButtonLabel"
    [isFullWidth]="true"
    [isPrimary]="true"
    (click)="reshareBtnClicked.emit($event)"
  ></dsmn8-button>
</div>
