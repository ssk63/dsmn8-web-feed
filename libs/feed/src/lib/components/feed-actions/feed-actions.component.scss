.feed-actions {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr;
  grid-gap: 20px;
  &__likes-btn,
  &__reshare-btn {
    button {
      width: 100%;
    }
  }
}
