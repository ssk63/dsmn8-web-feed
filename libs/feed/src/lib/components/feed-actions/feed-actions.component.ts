import { Component, Input, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'dsmn8-feed-actions',
  templateUrl: './feed-actions.component.html',
  styleUrls: ['./feed-actions.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class FeedActionsComponent {
  @Input() likesButtonLabel: string | undefined;
  @Input() reshareButtonLabel: string | undefined;
}
