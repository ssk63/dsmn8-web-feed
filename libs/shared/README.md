# shared

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared` to execute the unit tests.

## Running lint

Run `nx lint shared` to execute the lint.

## Running Storybook

Run `nx run shared:storybook` to execute the storybook.

## Running Cypress E2E

Run ` nx run shared-e2e:e2e --watch=true` to execute the storybook.