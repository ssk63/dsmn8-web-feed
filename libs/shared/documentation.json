{
    "pipes": [],
    "interfaces": [],
    "injectables": [],
    "guards": [],
    "interceptors": [],
    "classes": [],
    "directives": [],
    "components": [
        {
            "name": "ButtonComponent",
            "id": "component-ButtonComponent-0b8aa04796c6fc1c32b026ef96a6da46b1bcace03f8f46e58d21b2b7f085ac92c9e0a9d91bd749839508a28d18fcdf73aea0a20862a4d72028b53a97da118160",
            "file": "libs/shared/src/lib/components/button/button.component.ts",
            "encapsulation": [
                "ViewEncapsulation.None"
            ],
            "entryComponents": [],
            "inputs": [],
            "outputs": [],
            "providers": [],
            "selector": "dsmn8-button",
            "styleUrls": [
                "./button.component.scss"
            ],
            "styles": [],
            "templateUrl": [
                "./button.component.html"
            ],
            "viewProviders": [],
            "inputsClass": [
                {
                    "name": "ariaLabel",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "line": 17,
                    "type": "string | undefined",
                    "decorators": []
                },
                {
                    "name": "isDisabled",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "line": 19,
                    "type": "boolean | undefined",
                    "decorators": []
                },
                {
                    "name": "isFullWidth",
                    "defaultValue": "false",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "line": 20,
                    "type": "boolean",
                    "decorators": []
                },
                {
                    "name": "isPrimary",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "line": 18,
                    "type": "boolean | undefined",
                    "decorators": []
                },
                {
                    "name": "label",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "line": 16,
                    "type": "string | undefined",
                    "decorators": []
                }
            ],
            "outputsClass": [
                {
                    "name": "clicked",
                    "defaultValue": "new EventEmitter<boolean>()",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "line": 21,
                    "type": "EventEmitter"
                }
            ],
            "propertiesClass": [],
            "methodsClass": [
                {
                    "name": "onBtnClick",
                    "args": [],
                    "optional": false,
                    "returnType": "void",
                    "typeParameters": [],
                    "line": 23,
                    "deprecated": false,
                    "deprecationMessage": ""
                }
            ],
            "deprecated": false,
            "deprecationMessage": "",
            "hostBindings": [],
            "hostListeners": [],
            "description": "",
            "rawdescription": "\n",
            "type": "component",
            "sourceCode": "import {\n  Component,\n  EventEmitter,\n  Input,\n  Output,\n  ViewEncapsulation,\n} from '@angular/core';\n\n@Component({\n  selector: 'dsmn8-button',\n  templateUrl: './button.component.html',\n  styleUrls: ['./button.component.scss'],\n  encapsulation: ViewEncapsulation.None,\n})\nexport class ButtonComponent {\n  @Input() label: string | undefined;\n  @Input() ariaLabel: string | undefined;\n  @Input() isPrimary: boolean | undefined;\n  @Input() isDisabled: boolean | undefined;\n  @Input() isFullWidth = false;\n  @Output() clicked = new EventEmitter<boolean>();\n\n  onBtnClick(): void {\n    this.clicked.emit(true);\n  }\n}\n",
            "assetsDirs": [],
            "styleUrlsData": [
                {
                    "data": "$primary: #e85427;\n$primary-light: #fb7147;\n$grey: #455a76;\n$white: #ffffff;\n\n.dsmn8-btn {\n  &--primary {\n    border-color: $primary;\n    background-color: $primary;\n    color: $white;\n    &:hover {\n      border-color: $primary-light;\n      background-color: $primary-light;\n    }\n  }\n  &--block {\n    width: 100%;\n  }\n  &[disabled] {\n    background-color: $grey;\n    border-color: $grey;\n    color: $white;\n    opacity: 0.38;\n    cursor: default;\n  }\n  border: solid 1px #eceef1;\n  border-radius: 10px;\n  background-color: $white;\n  color: #494949;\n  padding: 12px 30px;\n  margin-bottom: 20px;\n  transition-duration: 0.5s;\n  text-align: center;\n  cursor: pointer;\n  min-width: 160px;\n  font-size: 16px;\n  font-weight: 700;\n}\n",
                    "styleUrl": "./button.component.scss"
                }
            ],
            "stylesData": "",
            "templateData": "<button\n  class=\"dsmn8-btn\"\n  [class.dsmn8-btn--primary]=\"isPrimary\"\n  [class.dsmn8-btn--block]=\"isFullWidth\"\n  [attr.aria-label]=\"ariaLabel\"\n  [disabled]=\"isDisabled\"\n  (click)=\"onBtnClick()\"\n  type=\"button\"\n  *ngIf=\"label\"\n>\n  {{ label }}\n</button>\n"
        }
    ],
    "modules": [],
    "miscellaneous": {
        "variables": [
            {
                "name": "Disabled",
                "ctype": "miscellaneous",
                "subtype": "variable",
                "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "deprecated": false,
                "deprecationMessage": "",
                "type": "",
                "defaultValue": "Template.bind({})"
            },
            {
                "name": "FullWidth",
                "ctype": "miscellaneous",
                "subtype": "variable",
                "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "deprecated": false,
                "deprecationMessage": "",
                "type": "",
                "defaultValue": "Template.bind({})"
            },
            {
                "name": "Primary",
                "ctype": "miscellaneous",
                "subtype": "variable",
                "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "deprecated": false,
                "deprecationMessage": "",
                "type": "",
                "defaultValue": "Template.bind({})"
            },
            {
                "name": "Template",
                "ctype": "miscellaneous",
                "subtype": "variable",
                "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "deprecated": false,
                "deprecationMessage": "",
                "type": "Story<ButtonComponent>",
                "defaultValue": "(args: ButtonComponent) => ({\n  props: {\n    ...args,\n    // clicked: action('log'),\n  },\n})"
            }
        ],
        "functions": [],
        "typealiases": [],
        "enumerations": [],
        "groupedVariables": {
            "libs/shared/src/lib/components/button/button.component.stories.ts": [
                {
                    "name": "Disabled",
                    "ctype": "miscellaneous",
                    "subtype": "variable",
                    "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "type": "",
                    "defaultValue": "Template.bind({})"
                },
                {
                    "name": "FullWidth",
                    "ctype": "miscellaneous",
                    "subtype": "variable",
                    "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "type": "",
                    "defaultValue": "Template.bind({})"
                },
                {
                    "name": "Primary",
                    "ctype": "miscellaneous",
                    "subtype": "variable",
                    "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "type": "",
                    "defaultValue": "Template.bind({})"
                },
                {
                    "name": "Template",
                    "ctype": "miscellaneous",
                    "subtype": "variable",
                    "file": "libs/shared/src/lib/components/button/button.component.stories.ts",
                    "deprecated": false,
                    "deprecationMessage": "",
                    "type": "Story<ButtonComponent>",
                    "defaultValue": "(args: ButtonComponent) => ({\n  props: {\n    ...args,\n    // clicked: action('log'),\n  },\n})"
                }
            ]
        },
        "groupedFunctions": {},
        "groupedEnumerations": {},
        "groupedTypeAliases": {}
    },
    "routes": [],
    "coverage": {
        "count": 0,
        "status": "low",
        "files": [
            {
                "filePath": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "type": "variable",
                "linktype": "miscellaneous",
                "linksubtype": "variable",
                "name": "Disabled",
                "coveragePercent": 0,
                "coverageCount": "0/1",
                "status": "low"
            },
            {
                "filePath": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "type": "variable",
                "linktype": "miscellaneous",
                "linksubtype": "variable",
                "name": "FullWidth",
                "coveragePercent": 0,
                "coverageCount": "0/1",
                "status": "low"
            },
            {
                "filePath": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "type": "variable",
                "linktype": "miscellaneous",
                "linksubtype": "variable",
                "name": "Primary",
                "coveragePercent": 0,
                "coverageCount": "0/1",
                "status": "low"
            },
            {
                "filePath": "libs/shared/src/lib/components/button/button.component.stories.ts",
                "type": "variable",
                "linktype": "miscellaneous",
                "linksubtype": "variable",
                "name": "Template",
                "coveragePercent": 0,
                "coverageCount": "0/1",
                "status": "low"
            },
            {
                "filePath": "libs/shared/src/lib/components/button/button.component.ts",
                "type": "component",
                "linktype": "component",
                "name": "ButtonComponent",
                "coveragePercent": 0,
                "coverageCount": "0/8",
                "status": "low"
            }
        ]
    }
}