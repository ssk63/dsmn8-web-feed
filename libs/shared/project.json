{
  "name": "shared",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/shared/src",
  "prefix": "dsmn8",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/shared"],
      "options": {
        "jestConfig": "libs/shared/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/shared/**/*.ts", "libs/shared/**/*.html"]
      }
    },
    "storybook": {
      "executor": "@storybook/angular:start-storybook",
      "options": {
        "port": 4401,
        "configDir": "libs/shared/.storybook",
        "browserTarget": "shared:build-storybook",
        "compodoc": true,
        "compodocArgs": ["-e", "json", "-d", "libs/shared"]
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "build-storybook": {
      "executor": "@storybook/angular:build-storybook",
      "outputs": ["{options.outputDir}"],
      "options": {
        "outputDir": "dist/storybook/shared",
        "configDir": "libs/shared/.storybook",
        "browserTarget": "shared:build-storybook",
        "compodoc": true,
        "compodocArgs": ["-e", "json", "-d", "libs/shared"]
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    }
  },
  "tags": []
}
