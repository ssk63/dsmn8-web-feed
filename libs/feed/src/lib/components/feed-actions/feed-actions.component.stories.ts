import { moduleMetadata, Story, Meta } from '@storybook/angular';
import { ButtonComponent } from '@dsmn8/shared';
import { likeBtnLabel, reshareBtnLabel } from '../../models';
import { FeedActionsComponent } from './feed-actions.component';

export default {
  title: 'Pages/Feed/FeedActionsComponent',
  component: FeedActionsComponent,
  decorators: [
    moduleMetadata({
      imports: [],
      declarations: [ButtonComponent],
    }),
  ],
  argTypes: {
    likesButtonLabel: {
      name: 'likesButtonLabel',
      description: 'Sets the like button text.',
      type: { name: 'string', required: true },
      control: {
        type: 'text',
      },
      table: {
        category: 'Inputs',
      },
    },
    isLiked: {
      name: 'isLiked',
      description: 'Sets the like button enable and disable',
      type: { name: 'boolean', required: true },
      control: {
        type: 'boolean',
      },
      table: {
        category: 'Inputs',
      },
    },
    reshareButtonLabel: {
      name: 'reshareButtonLabel',
      description: 'Sets the reshare button text.',
      type: { name: 'string', required: true },
      control: {
        type: 'text',
      },
      table: {
        category: 'Inputs',
      },
    },
    likesBtnClicked: {
      description: 'Event that is emitted when the like button is clicked.',
      action: 'likesBtnClicked',
      table: {
        category: 'Outputs',
      },
    },
    reshareBtnClicked: {
      description: 'Event that is emitted when the reshare button is clicked.',
      action: 'reshareBtnClicked',
      table: {
        category: 'Outputs',
      },
    },
  },
} as Meta<FeedActionsComponent>;

const Template: Story<FeedActionsComponent> = (args: FeedActionsComponent) => ({
  props: args,
});

export const FeedActions = Template.bind({});
FeedActions.args = {
  likesButtonLabel: likeBtnLabel,
  reshareButtonLabel: reshareBtnLabel,
  isLiked: false
};
