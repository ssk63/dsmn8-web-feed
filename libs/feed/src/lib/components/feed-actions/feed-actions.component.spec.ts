import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FeedActionsComponent } from './feed-actions.component';
import { ButtonComponent } from '@dsmn8/shared';
import { By } from '@angular/platform-browser';

describe('FeedActionsComponent', () => {
  let component: FeedActionsComponent;
  let fixture: ComponentFixture<FeedActionsComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FeedActionsComponent, ButtonComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(FeedActionsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should render likes button when the likesButtonLabel is exists', () => {
    // arrange
    component.likesButtonLabel = 'Likes Button';
    // act
    fixture.detectChanges();
    const likesButton = fixture.debugElement.query(
      By.css('.feed-actions__likes-btn')
    ).nativeElement;
    // assert
    expect(likesButton).toMatchSnapshot();
  });

  test('should render reshare button when the reshareButtonLabel is exists', () => {
    // arrange
    component.reshareButtonLabel = 'Likes Button';
    // act
    fixture.detectChanges();
    const reshareButton = fixture.debugElement.query(
      By.css('.feed-actions__reshare-btn')
    ).nativeElement;
    // assert
    expect(reshareButton).toMatchSnapshot();
  });

  test('should emit event when the likes button is clicked', () => {
    // arrange
    jest.spyOn(component.likesBtnClicked, 'emit');
    component.likesButtonLabel = 'Likes Button';
    // act
    fixture.detectChanges();
    const likesButton = fixture.debugElement.query(
      By.css('.feed-actions__likes-btn')
    ).nativeElement;
    likesButton.click();
    // assert
    expect(component.likesBtnClicked.emit).toBeCalled();
  });

  test('should emit event when the reshare button is clicked', () => {
    // arrange
    jest.spyOn(component.reshareBtnClicked, 'emit');
    component.reshareButtonLabel = 'Reshare Button';
    // act
    fixture.detectChanges();
    const reshareButton = fixture.debugElement.query(
      By.css('.feed-actions__reshare-btn')
    ).nativeElement;
    reshareButton.click();
    // assert
    expect(component.reshareBtnClicked.emit).toBeCalled();
  });
});
