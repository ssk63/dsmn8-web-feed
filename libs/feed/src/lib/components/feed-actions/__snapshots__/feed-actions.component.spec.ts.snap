// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedActionsComponent should render likes button when the likesButtonLabel is exists 1`] = `
<dsmn8-button
  class="feed-actions__likes-btn"
>
  <button
    class="dsmn8-btn dsmn8-btn--primary dsmn8-btn--block"
    type="button"
  >
     Likes Button

  </button>
  
</dsmn8-button>
`;

exports[`FeedActionsComponent should render reshare button when the reshareButtonLabel is exists 1`] = `
<dsmn8-button
  class="feed-actions__reshare-btn"
>
  <button
    class="dsmn8-btn dsmn8-btn--primary dsmn8-btn--block"
    type="button"
  >
     Likes Button

  </button>
  
</dsmn8-button>
`;
